editor/src/clj/editor/gui.clj

```diff
@@ -567,6 +535,38 @@
       (when emit-warnings?
         (validate-contains :warning "layer '%s' from template scene does not exist in the scene - will use layer of parent" :layer node-id layer-names layer)))))
 
+(defn- overridden-fields [gui-node]
+  (->> gui-node
+       (gt/overridden-properties)
+       (keys)
+       (keep prop-key->prop-index)
+       (sort)
+       (vec)))
+
+(g/defnk produce-gui-base-node-msg [_this type custom-type child-index position rotation scale id generated-id color alpha inherit-alpha enabled layer parent]
+  ;; Warning: This base output or any of the base outputs that derive from it
+  ;; must not be cached due to overridden-fields reliance on _this. Only the
+  ;; node-msg outputs of concrete nodes may be cached. In that case caching is
+  ;; fine since such an output will have explicit dependencies on all the
+  ;; property values it requires.
+  (cond-> {:type type
+           :custom-type custom-type
+           :child-index child-index
+           :template-node-child false
+           :overridden-fields (overridden-fields _this)
+           :position (v3->v4 position)
+           :rotation (clj-quat->euler-v4 rotation)
+           :scale (v3->v4 scale)
+           :id (if (g/node-override? _this) generated-id id)
+           :color color ; TODO: Not used by template (forced to [1.0 1.0 1.0 1.0]). Move?
```

Comment:
These TODOs were discovered when digging through the existing implementation. We might want to refactor the properties in the future, but for now we prioritize minimizing file format changes.